packages/react/src/components/UIShell/HeaderPanel.js

```diff
@@ -27,14 +31,36 @@ const HeaderPanel = React.forwardRef(function HeaderPanel(
     'aria-label': ariaLabel,
     'aria-labelledby': ariaLabelledBy,
   };
+  const controlled = useRef(expanded !== undefined).current;
+  const [expandedState, setExpandedState] = useState(expanded);
+  const expandedProp = controlled ? expanded : expandedState;
 
   const className = cx(`${prefix}--header-panel`, {
-    [`${prefix}--header-panel--expanded`]: expanded,
+    [`${prefix}--header-panel--expanded`]: expandedProp,
     [customClassName]: !!customClassName,
   });
 
+  const eventHandlers = {};
+
+  if (addFocusListeners) {
+    eventHandlers.onKeyDown = (event) => {
+      if (match(event, keys.Escape)) {
+        setExpandedState(false);
+        onHeaderPanelFocus();
+        if (href) {
+          window.location.href = href;
+        }
```

Comment:
now that i'm thinking about this I don't think it makes sense for the HeaderPanel to redirect to the main content when the headerPanel closes. I think this is an interaction that makes sense for the side rail but not necessarily here 🤔

Reply:
I agree. Now that the menu can't be open on focus, this prop is not necessary